plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.test.ts

```diff
@@ -127,12 +128,45 @@ describe('ArgoCD service', () => {
       });
     });
 
+    it('should get revision data when the Argo CD instance is configured with a trailing slash', async () => {
+      const expectedResp = {
+        author: 'testuser',
+        date: '2023-03-20T18:44:10Z',
+        message: 'Update README.md',
+      }
+      
+      fetchMock.mockResponseOnce(JSON.stringify(expectedResp));
+
+      const name = 'testApp'
+      const token = 'testToken'
+      const revisionId = '15db63ac922a920f388bd841912838ae4d126317'
+
+      const resp = await argoService.getRevisionData(
+        'https://argoinstance1.com/',
+        { name },
+        token,
+        revisionId,
+      );
+
+      expect(fetchMock).toHaveBeenCalledWith(
+        expect.stringMatching(`https://argoinstance1.com/api/v1/applications/${name}/revisions/${revisionId}/metadata`),
+        expect.objectContaining({
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${token}`,
+          }
+        }),
+      );
+
+      expect(resp).toStrictEqual(expectedResp);
```

Comment:
Actually this assertion is a bit pointless or at least out of place. You're confirming that `fetchMock` works essentially. They have their own tests for that. I'd just remove this assertion and restore the inline call you had (removing `expectedResp`).

Reply:
Not entirely - it's confirming we got the correct response from `getRevisionData`. The test shouldn't assume that the value returned from `getRevisionData` is identical to the data returned from the mock, since the internals of `getRevisionData` _could_ change at some point. However the prior test can handle this, so maybe this test can simply be:

```js
expect(fetchMock.mock.calls[0][0]).toBe(``https://argoinstance1.com/api/v1/applications/${name}/revisions/${revisionId}/metadata``)
```

And call it a day?

Reply:
@evanshortiss Thoughts on using [`it.each`](https://jestjs.io/docs/api#testconcurrenteachtablename-fn-timeout) for the tests to make it apparent that they are identical besides a singular input/character?